pkg/state/appender.go

```diff
@@ -698,16 +739,18 @@ func (a *txAppender) appendBlock(params *appendBlockParams) error {
 			stateActionsCounterInBlock:       stateActionsCounterInBlockValidation,
 			currentMinerPK:                   params.block.GeneratorPublicKey,
 		}
-		if err := a.appendTx(tx, appendTxArgs); err != nil {
-			return err
+		txSnapshots, errAppendTx := a.appendTx(tx, appendTxArgs)
+		if errAppendTx != nil {
+			return errAppendTx
 		}
+		blockSnapshots.TxSnapshots = append(blockSnapshots.TxSnapshots, txSnapshots.regular)
```

Comment:
On my mind It's better to create an `append` function or method for `proto.BlockSnapshot`.